### DISCUSSION TITLE
VSCode extension and IntelliSense

### DISCUSSION THREAD

**Discussion by lonix1** [cake-build/cake#3929] [Q&A] [answered]
I posted a bug report and repro [here](https://github.com/cake-build/cake-vscode/issues/693) but then discovered this org-level discussion, so I'm unsure if I put it in the right place.

Is the latest cake-vscode extension compatible with the latest omnisharp-vscode extension (which no longer relies on mono) on linux? I suspect it is not compatible as I've tried everything (see above link) without success.

**Comment by lonix1** [comment]
**I think this is a bug**

As mentioned in that issue linked above, I'm using linux and the latest vscode. So it uses the "modern" omnisharp config, without mono.

I wanted to see whether it works with the mono build:
- I added an `omnisharp.json` file which contains: `{ "SDK": { "Version": "5.0.408" } }`
- I added to the `.vscode/settings.json` file: `"omnisharp.useModernNet": false,`
- I restarted vscode

...and IntelliSense worked. Obviously I don't want to use the old O#/mono, but it shows there is a compatibility problem.

With the old mono o#, the log now has this:
````
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/home/foo/project'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Found 1 Cake files.
````

## Is this a bug in the vscode-cake extension?

**Comment by denisz1** [comment]
This problem I have also.

**Is someone can confirm VSCode extension work on Linux and .NET 6???**

If not compatible, it is sad but OK. :smile: But I spend so much time trying to fix this - I like to know if to continue, or give up?!

Please please advise me? :pray:

**Comment by nils-a** [comment, marked as answer]
Due to some constraints, [OmniSharp][1] (the basis for the [VSCode C# extension][2], which also includes support for Cake) currently does not ship the support for Cake in the `net6` versions.

So you have two options here:
* set `"omnisharp.useModernNet": false` in the vscode settings. This will not use the `net6` version of OmniSharp and fall back the mono/.NET Framework versions.

* set `"omnisharp.path": "latest"` in the vscode settings. This will always use the latest beta version of OmniSharp (at the time of writing, this is `1.39.2-beta.4`). OmniSharp does (will?) ship support for Cake in the `net6` versions starting with `1.39.1` (which is currently available for download, but not used in the c# extension for vs code.) 

  [1]: https://github.com/OmniSharp/omnisharp-roslyn
  [2]: https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp

**Reply by bjorkstromm** [reply]
Starting of `1.39.2` OmniSharp .NET6 builds will include Cake bits. Until that is released and a new C# extension is released, I'd recommend the following settings in VS Code:


```jsonc
"omnisharp.useModernNet": true // <-- This will use .NET 6 builds
"omnisharp.path": "latest" // <-- This will install lates CI builds of OmniSharp
```

You can also use `1.39.2-beta.39` instead of `latest` as this is the latest working in time of writing.

**Reply by lonix1** [reply]
Thanks. Just keep in mind that `latest` means it will download every new release, even prerelease.

Side note: that was a problem before, but omnisharp hasn't been updated in a long time now, and I'm worried it's basically dead. I wonder what'll happen once .net7 is released next month.
